pkg/controller/jaeger/ingress.go

```diff
@@ -0,0 +1,59 @@
+package jaeger
+
+import (
+	"context"
+
+	log "github.com/sirupsen/logrus"
+	"k8s.io/api/extensions/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/jaegertracing/jaeger-operator/pkg/apis/io/v1alpha1"
+	"github.com/jaegertracing/jaeger-operator/pkg/inventory"
+)
+
+func (r *ReconcileJaeger) applyIngresses(jaeger v1alpha1.Jaeger, desired []v1beta1.Ingress) error {
+	opts := client.MatchingLabels(map[string]string{
+		"app.kubernetes.io/instance":   jaeger.Name,
+		"app.kubernetes.io/managed-by": "jaeger-operator",
+	})
+	list := &v1beta1.IngressList{}
+	if err := r.client.List(context.Background(), opts, list); err != nil {
+		return err
+	}
+
+	inv := inventory.ForIngresses(list.Items, desired)
+	for _, d := range inv.Create {
+		log.WithFields(log.Fields{
+			"namespace": jaeger.Namespace,
+			"instance":  jaeger.Name,
+			"cronjob":   d.Name,
+		}).Debug("creating cronjob")
+		if err := r.client.Create(context.Background(), &d); err != nil {
+			return err
+		}
+	}
+
+	for _, d := range inv.Update {
+		log.WithFields(log.Fields{
+			"namespace": jaeger.Namespace,
+			"instance":  jaeger.Name,
+			"cronjob":   d.Name,
+		}).Debug("updating cronjob")
+		if err := r.client.Update(context.Background(), &d); err != nil {
+			return err
+		}
+	}
+
+	for _, d := range inv.Delete {
```

Comment:
Just wondering if ingresses need to be handled similar to deployments - i.e. wait for new and updated to be stable before deleting old?

Reply:
Just gave it a try, and ingresses are most likely not subject to many changes at all, as they point to services, like `simple-prod-query`:

```yaml
  spec:
    backend:
      serviceName: simple-prod-query
      servicePort: 16686
```

IIRC, we have only one ingress, and it's installed in all cases (except when running in OpenShift), so, I don't think we currently have the case where we have a new/updated ingress and an old one to be removed.